single-kernel/mol_dyn.cpp

```diff
@@ -0,0 +1,143 @@
+#include "common.h"
+#include <iostream>
+
+//using namespace cl::sycl;
+namespace s = cl::sycl;
+class MolecularDynamicsKernel;
+
+class MolecularDynamicsBench
+{
+protected:    
+    std::vector<s::float4> input;
+    std::vector<s::float4> output;
+    std::vector<int> neighbour;
+	int neighCount;
+	int cutsq;
+	int lj1;
+	int lj2;
+	int inum;
+    BenchmarkArgs args;
+
+public:
+  MolecularDynamicsBench(const BenchmarkArgs &_args) : args(_args) {}
+  
+  void setup() {      
+    // host memory allocation and initialization
+	neighCount = 15;
+	cutsq = 50;
+	lj1 = 20;
+	lj2 = 0.003f;
```

Comment:
clang warns `warning: implicit conversion from 'float' to 'int' changes value from 0.003 to 0 [-Wliteral-conversion]` because lj2 is actually defined as an int. Are you sure this is what you want?

Reply:
This was the case with the original OpenCL code. I am verifying it and will fix it.